File: android_app/app/src/main/java/com/snoy/apptoapp/android/model/RealtimeQuotRepo.kt
package com.snoy.apptoapp.android.model

import kotlinx.coroutines.flow.Flow

interface RealtimeQuotRepo {
    fun observeRealtimeQuote(): Flow<QuotItem?>
    suspend fun updateQuot(): QuotItem?
    suspend fun toggleRealtimeQuote(enable: Boolean)
}

File: ios_app/ios_app/model/RealtimeQuotRepo.swift
//
//  RealtimeQuotRepo.swift
//  ios_app
//
//  Created by Snoy Kuo on 2021/6/15.
//

import Combine

protocol RealtimeQuotRepo {
    func observeRealtimeQuote() -> PassthroughSubject<QuotItem?,Never>
    func updateQuot() -> Future<QuotItem?, Never>
    func toggleRealtimeQuote(enable: Bool)
}
